### PR TITLE
Comments block: Try client-side form submission [new store API]

### PR DESCRIPTION
**Pull request by ockham**
## What?
Attempt updating #53737 to use the new Store API.

## Why?
As a preparatory step to working on fetching required block styles and scripts upon comment submission.

## How?
By looking at [this diff](https://github.com/WordPress/gutenberg/pull/56764/files) and [this](https://github.com/WordPress/gutenberg/pull/56143/files).

### CODE REVIEW

**Comment by ockham** [comment]
Currently failing on the frontend (if a Comments block is present):

image

**Comment by ockham** [comment]
👋 @luisherranz Any advice on the above [error](https://github.com/WordPress/gutenberg/pull/56984#issuecomment-1852323568)? 😬

**Comment by ockham** [comment]
Thank you @DAreRodz for getting this to work!

File: packages/block-library/src/post-comments-form/view.js [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+			const { loadingText, submitText } = state;
+			return isSubmitting ? loadingText : submitText;
+		},
+		get displayCancelReply() {
+			const { formSlot } = getContext();
+			return formSlot ? undefined : 'none';
+		},
+	},
+	actions: {
+		submit: async ( event ) => {
```
Comment (luisherranz) [inline packages/block-library/src/post-comments-form/view.js:36]: This needs to use `*` instead of `async` and `yield` instead of `await` 🙂

**Comment by ockham** [comment]
I've carried the changes back over to #53737. Closing this PR.

**Pull request closed without merging**